### PR TITLE
Dependencies updates and TYPO3 upgrade to 8.7

### PR DESCRIPTION
**Pull request by dkd-ebert**
- rubocop upgrade to 0.50
- ruby upgrade to 2.2
- bundler upgrade
- rake upgrade
- dkdeploy-test_environment upgrade to 2.0
- dkdeploy-core upgrade 9.0
- Add dependencies for `cucumber`, `rubocop`, `aruba` and `mysql2
- upgrade vagrant box to `ubuntu/xenial64`
- update TYPO3 8.7.8
- moving tasks from step_definitions/typo3.rb to deploy.rb

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,5 +8,18 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
 - first public release
 
+### Changed
+
+- rubocop upgrade to 0.50
+- ruby upgrade to 2.2
+- bundler upgrade
+- rake upgrade
+- dkdeploy-test_environment upgrade to 2.0
+- dkdeploy-core upgrade 9.0
```
Comment (websi) [inline CHANGELOG.md]: `dkdpeloy-php` is missing

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,5 +8,18 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+- Add dependencies for `cucumber`, `rubocop`, `aruba` and `mysql2
+- upgrade vagrant box to `ubuntu/xenial64`
+- update TYPO3 8.7.8
+- moving tasks from step_definitions/typo3.rb to deploy.rb
```
Comment (websi) [inline CHANGELOG.md]: This was only for the testing environment. Please add this information to changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,5 +8,18 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+
+- rubocop upgrade to 0.50
+- ruby upgrade to 2.2
+- bundler upgrade
+- rake upgrade
+- dkdeploy-test_environment upgrade to 2.0
+- dkdeploy-core upgrade 9.0
+- Add dependencies for `cucumber`, `rubocop`, `aruba` and `mysql2
+- upgrade vagrant box to `ubuntu/xenial64`
+- update TYPO3 8.7.8
```
Comment (websi) [inline CHANGELOG.md]: This release should support 8.7.x. Not only 8.7.8.

File: .gitignore [hunk 4]
```diff
@@ -18,3 +18,10 @@ tmp
 .vagrant
 .lock
 Gemfile.lock
+spec/fixtures/application/htdocs/typo3_src
+spec/fixtures/application/htdocs/typo3conf/ext
+spec/fixtures/application/htdocs/index.php
+spec/fixtures/application/htdocs/typo3
+spec/fixtures/application/htdocs/bin
```
Comment (websi) [inline .gitignore]: Is this really needed?
Dependencies should be installed within directory `tmp/aruba`

File: lib/capistrano/dkdeploy/typo3_cms.rb [hunk 5]
```diff
@@ -27,7 +27,8 @@
       TERM: 'screen-256color'
     }
     # Path to typo3_console. Relative path to typo3_console script
-    set :path_to_typo3_console, 'bin/typo3cms'
+    set :path_to_typo3_console, File.join('bin', 'typo3cms')
+    set :path_to_typo3_cli, File.join('bin', 'typo3')
```
Comment (websi) [inline lib/capistrano/dkdeploy/typo3_cms.rb]: Task `typo3:cms:cli:upload_wrapper` should use the new configuration option or delete task.

File: lib/capistrano/dkdeploy/typo3_cms.rb [hunk 6]
```diff
@@ -27,7 +27,8 @@
       TERM: 'screen-256color'
```
Comment (websi) [inline lib/capistrano/dkdeploy/typo3_cms.rb]: Please remove `TYPO3_COMPOSER_AUTOLOAD`. Is not needed for TYPO3 8.7.x

File: .gitignore [hunk 7]
```diff
@@ -18,3 +18,10 @@ tmp
...
+
+
```
Comment (websi) [inline .gitignore]: Please add `ubuntu-xenial-16.04-cloudimg-console.log`

**Comment by dkd-ebert** [comment]
I commited the changes to the feedback, please check again

File: lib/capistrano/dkdeploy/typo3_cms.rb [hunk 8]
```diff
@@ -21,13 +21,13 @@
     set :create_caretaker_instance_keys_path, -> { File.join(shared_path, 'config', 'create_caretaker_instance_keys.php') }
     set :caretaker_instance_keys_path, -> { File.join(shared_path, 'config', 'caretaker_instance_keys.php') }
 
-    # Use custom Composer autoload
     set :typo3_environment_cli, {
-      TYPO3_COMPOSER_AUTOLOAD: 1,
+      CUSTOM_VARIABLE: 1,
```
Comment (websi) [inline lib/capistrano/dkdeploy/typo3_cms.rb]: Please remove `CUSTOM_VARIABLE`.
If needed for tests, please move it to test application.

**Comment by dkd-ebert** [comment]
moved CUSTOM_VARIABLE to fixtures

**Pull request merged**